twitter-workitem/src/main/java/org/jbpm/process/workitem/twitter/UpdateStatusWorkitemHandler.java

```diff
@@ -0,0 +1,129 @@
+import org.jbpm.process.workitem.core.util.WidParameter;
+
+import org.kie.api.runtime.process.WorkItem;
+import org.kie.api.runtime.process.WorkItemManager;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import twitter4j.StatusUpdate;
+import twitter4j.Twitter;
+
+@Wid(widfile = "TwitterUpdateStatus.wid", name = "TwitterUpdateStatus",
+        displayName = "TwitterUpdateStatus",
+        defaultHandler = "mvel: new org.jbpm.process.workitem.twitter.UpdateStatusWorkitemHandler()",
+        parameters = {
+                @WidParameter(name = "Message"),
+                @WidParameter(name = "Media")
+        },
+        mavenDepends = {
+                @WidMavenDepends(group = "org.twitter4j", artifact = "twitter4j-core", version = "4.0.6")
+        })
+public class UpdateStatusWorkitemHandler extends AbstractLogOrThrowWorkItemHandler {
+
+    private static final Logger logger = LoggerFactory.getLogger(UpdateStatusWorkitemHandler.class);
+
+    private TwitterAuth auth = new TwitterAuth();
+
+    private String consumerKey;
+    private String consumerSecret;
+    private String accessKey;
+    private String accessSecret;
+    private StatusUpdate statusUpdate;
+
+    public UpdateStatusWorkitemHandler(String consumerKey,
+                                       String consumerSecret,
+                                       String accessKey,
+                                       String accessSecret) {
+        this.consumerKey = consumerKey;
+        this.consumerSecret = consumerSecret;
+        this.accessKey = accessKey;
+        this.accessSecret = accessSecret;
+    }
+
+    public void executeWorkItem(WorkItem workItem,
+                                WorkItemManager workItemManager) {
+
+        String statusMessage = (String) workItem.getParameter("Message");
+
+        // media is optional
+        Document statusMedia = null;
+        if (workItem.getParameter("Media") != null) {
+            statusMedia = (Document) workItem.getParameter("Media");
+        }
+
+        if (StringUtils.isNotEmpty(statusMessage)) {
+
+            try {
+                Twitter twitter = auth.getTwitterService(this.consumerKey,
+                                                         this.consumerSecret,
+                                                         this.accessKey,
+                                                         this.accessSecret);
+
+                statusUpdate = new StatusUpdate(statusMessage);
+                if (statusMedia != null) {
+
+                    File tempMediaFile = File.createTempFile(FilenameUtils.getBaseName(statusMedia.getName()),
+                                                             "." + FilenameUtils.getExtension(statusMedia.getName()));
+
+                    FileOutputStream fos = new FileOutputStream(tempMediaFile);
+                    fos.write(statusMedia.getContent());
+                    fos.close();
+
+                    statusUpdate.setMedia(tempMediaFile);
```

Comment:
maybe use setMedia(java.lang.String name, java.io.InputStream body)  instead so no temp file is needed